tests/versioned/express-segments-tests.js

```diff
@@ -0,0 +1,661 @@
+
+      const expectedSegments = [{
+        name: `${batchTransactionPrefix}/${expectedQuery1Name}/${expectedQuery2Name}`,
+        children: [{
+          name: 'Expressjs/Router: /',
+          children: [{
+            name: 'Nodejs/Middleware/Expressjs/<anonymous>',
+            children: [
+              {
+                name: operationPart1,
+                children: [
+                  {
+                    name: 'resolve: library',
+                    children: [{
+                      name: 'timers.setTimeout',
+                      children: [{
+                        name: 'Callback: <anonymous>'
+                      }]
+                    }]
+                  },
+                  { name: 'resolve: library.books' },
+                  { name: 'resolve: library.books.title'},
+                  { name: 'resolve: library.books.author' },
+                  { name: 'resolve: library.books.author.name' }
+                ]
+              },
+              {
+                name: operationPart2,
+                children: [{
+                  name: 'resolve: addThing',
+                  children: [{
+                    name: 'timers.setTimeout',
+                    children: [{
+                      name: 'Callback: namedCallback'
+                    }]
+                  }]
+                }]
+              }
+            ]
+          }]
+        }]
+      }]
+
+      t.segments(transaction.trace.root, expectedSegments)
+    })
+
+    executeQueryBatch(serverUrl, queries, (err, result) => {
+      t.error(err)
+      checkResult(t, result, () => {
+        t.equal(result.length, 2)
+
+        t.end()
+      })
+    })
+  })
+
+  // there will be no document/AST nor resolved operation
+  t.test('when the query cannot be parsed, should have operation placeholder', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const invalidQuery = `query {
+      libraries {
+        books {
+          title
+          author {
+            name
+          }
+        }
+      }
+    ` // missing closing }
+
+    helper.agent.on('transactionFinished', (transaction) => {
+      t.equal(
+        transaction.name,
+        'WebTransaction/apollo-server/*'
+      )
+
+      const expectedSegments = [{
+        name: 'WebTransaction/apollo-server/*',
+        children: [{
+          name: 'Expressjs/Router: /',
+          children: [{
+            name: 'Nodejs/Middleware/Expressjs/<anonymous>',
+            children: [{
+              name: UNKNOWN_OPERATION_PLACEHOLDER
+            }]
+          }]
+        }]
+      }]
+
+      t.segments(transaction.trace.root, expectedSegments)
+    })
+
+    executeQuery(serverUrl, invalidQuery, (err, result) => {
+      t.error(err)
+
+      t.ok(result)
+      t.ok(result.errors)
+      t.equal(result.errors.length, 1) // should have one parsing error
+
+      const [parseError] = result.errors
+      t.equal(parseError.extensions.code, 'GRAPHQL_PARSE_FAILED')
```

Comment:
Should we have another test to check that parseError.extensions.code exists to protect against api breaking?  I'm assuming `extensions.code` is part of Apollo.

Reply:
hmm. I don't think we are doing anything related to errors that I'd expect to change fundamentals.

Perhaps? Although, I guess this sort of covers that anyways.

Reply:
oh word, yeah it is covered on line 585.  please disregard :)